src/gantt.ts

```diff
@@ -775,30 +788,68 @@ module powerbi.extensibility.visual {
                     } while (task.daysOffList.length && datesDiff - DaysInAWeekend > DaysInAWeek);
                 }
 
-                task.tooltipInfo = Gantt.getTooltipInfo(task, host.locale, formatters, settings.general.durationUnit);
+                task.tooltipInfo = Gantt.getTooltipInfo(task, host.locale, formatters, durationUnit);
 
                 tasks.push(task);
             });
 
             return tasks;
         }
 
+        private static downgradeDurationUnit(durationUnit: string): string {
+            let durationUnitTypeIndex = GanttDurationUnitType.indexOf(durationUnit);
+            durationUnit = (durationUnitTypeIndex !== -1 && durationUnit) || "day";
+            for (let i = 2; i > 0; i--) {
+                if (durationUnitTypeIndex - i > -1) {
+                    durationUnit = GanttDurationUnitType[durationUnitTypeIndex - i];
+                    break;
+                }
+            }
+
+            return durationUnit;
+        }
+
         private static transformExtraDuration(
             durationUnit: string,
-            extraDuration: number): number {
+            duration: number): number {
             switch (durationUnit) {
                 case "hour":
-                    return HoursInADay * extraDuration;
+                    return HoursInADay * duration;
 
                 case "minute":
-                    return MinutesInADay * extraDuration;
+                    return MinutesInADay * duration;
 
                 case "second":
-                    return SecondsInADay * extraDuration;
+                    return SecondsInADay * duration;
 
                 default:
-                    return extraDuration;
+                    return duration;
+            }
+
+        }
+
+        private static transformDuration(
+            duration: number,
+            newDurationUnit: string,
+            stepDurationTransformation: number): number {
+            let transformedDuration: number = duration;
+            switch (newDurationUnit) {
+                case "hour":
```

Comment:
Why don't you use any Enum instead of hard-coded strings?

Reply:
Done